compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -1858,11 +1862,23 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
             }
             GeneratorInteriorOrUpvar::Upvar(upvar_span) => {
                 let mut span = MultiSpan::from_span(upvar_span);
-                span.push_span_label(
-                    upvar_span,
-                    format!("has type `{}` which {}", target_ty, trait_explanation),
-                );
+                match target_ty.kind() {
+                    ty::Ref(_, ref_ty, _) => {
+                        span.push_span_label(
+                            upvar_span,
+                            // TODO: we need the equivalent of `trait_explanation` for the `ref_ty`
```

Comment:
need to figure out how to get the equivalent of `trait_explanation` for the `ref_ty` -- i.e., the type of the reference. 

When `target_ty` is of type `&T` and `T: Sync` is not satisfied, we need to be able to say 

`refers to type 'T' which 'is not Sync'`